### PR TITLE
Set Down-link Band Menu

### PR DESCRIPTION
**Pull request by tsteinholz**
This pull request includes #3 , but they're both good to go :100:

### CODE REVIEW

File: projects/network_tester/src/ui_task.c [hunk 1]
```diff
@@ -657,6 +673,35 @@ static void ui_load_menu(void)
             screen[(menu_pos%3)+2][0] = CURSOR_GLYPH;
             xTaskNotifyGive(s_screen_task_handle);
             break;
+        case SET_DL_BAND_MENU:
+            menu_pos_floor = (menu_pos/3) * 3;
+            llabs_dl_band_cfg_t dl_band_cfg;
+            int8_t active_pos;
+            if (ll_dl_band_cfg_get(&dl_band_cfg) != 0) {
+                if (dl_band_cfg.band_edge_lower == FCC.band_edge_lower)              active_pos = 0;
```
Comment (awelkiell) [inline projects/network_tester/src/ui_task.c]: Use parentheses around each statement. A `switch` statement might be easier.

File: projects/network_tester/src/supervisor.c [hunk 2]
```diff
@@ -783,7 +783,7 @@ static void sup_ll_after_reset_state()
         s_is_drive_mode = sup_drive_mode_init();
 
         // Set DL Band config to default
-        sup_dl_band_config(s_is_drive_mode);
+        // sup_dl_band_config(s_is_drive_mode);
```
Comment (awelkiell) [inline projects/network_tester/src/supervisor.c]: Should we remove this variable altogether?

File: projects/network_tester/src/ui_task.c (hunk 1 above)
Comment (tsteinholz) [inline projects/network_tester/src/ui_task.c]: C doesn't let us use a switch here :-1: @awelkiell 

`error: case label does not reduce to an integer constant`

http://stackoverflow.com/questions/24822887/use-const-struct-member-in-switch-case-statement

File: projects/network_tester/inc/supervisor.h [hunk 3]
```diff
@@ -21,6 +21,12 @@ typedef struct
 
 /*********************************************************************/
 /*****PUBLIC DEFS*****************************************************/
+// downlink band configs
+extern const llabs_dl_band_cfg_t DL_BAN_FCC;  // USA / Mexico
+extern const llabs_dl_band_cfg_t DL_BAN_BRA;  // Brazil
+extern const llabs_dl_band_cfg_t DL_BAN_AUS;  // Australia
+extern const llabs_dl_band_cfg_t DL_BAN_NZL;  // New Zealand
+extern const llabs_dl_band_cfg_t DL_BAN_ETSI; // Europe
```
Comment (tsteinholz) [inline projects/network_tester/inc/supervisor.h]: Any file that includes supervisor.h can view the dl_band_cfg for the following localization settings. This might be worth moving to the ifc at some point.

File: projects/network_tester/src/supervisor.c [hunk 4]
```diff
@@ -595,45 +588,23 @@ static bool sup_drive_mode_init(void)
...
-            cfg.chan_step_size == 100 &&
-            cfg.chan_step_offset < 50)
-        {
-            return true;
-        }
-        else
-        {
-            return false;
-        }
+        return cfg.chan_step_size == 100 && cfg.chan_step_offset < 50;
```
Comment (tsteinholz) [inline projects/network_tester/src/supervisor.c]: The band limits should already be set at this point. Drive-by Mode should only be defined by the channel step and the offset.

File: projects/network_tester/src/supervisor.c [hunk 5]
```diff
@@ -782,9 +753,6 @@ static void sup_ll_after_reset_state()
         // Set drive mode by reading dl_band_cfg
         s_is_drive_mode = sup_drive_mode_init();
 
-        // Set DL Band config to default
-        sup_dl_band_config(s_is_drive_mode);
-
         // update display with module info
```
Comment (tsteinholz) [inline projects/network_tester/src/supervisor.c]: We do not want to set the dl band to the default, we want the user's localization settings to be saved. `s_is_drive` is used by the supervisor to know when the network tester is in drive mode (shouldn't be removed). The network tester should still be configured with drive mode in the correct localization when the network tester is booting.

File: projects/network_tester/src/ui_task.c [hunk 6]
```diff
@@ -657,6 +668,62 @@ static void ui_load_menu(void)
...
+		    active_pos = CURSOR_LINE_4;
+		}
+		else if (dl_band_cfg.band_edge_lower == DL_BAN_ETSI.band_edge_lower)
+		{
+		    active_pos = CURSOR_LINE_5;
+		}
+		else
+		{
+		    active_pos = -1;
+		}
```
Comment (tsteinholz) [inline projects/network_tester/src/ui_task.c]: The only way I could make this a switch statement is if I set up the downlink configurations in an array that corresponded to a typedef enum that corresponds the name with the array index. This is because switch cases must be constant and cannot be any variable name (even if its constant? :-1: ) I don't think that is worth it though.

Comment (awelkiell) [inline projects/network_tester/src/ui_task.c]: Agreed
